# Introduction to analysis of next (NGS) and third generation sequencing data

## Overall Aim
Basic knowledge in Genetics and and Computional Biology is needed to run genetic analyses and interpret results. The goal of this course is to give students insights in both areas and present them a blueprint for large-scale studies.  

## Location & Time
* April 24 - April 27 2022 
* Start: April 24 9:00am
* Location: Computerraum C1 (FP301200) Fritz Pregl Strasse 3, 1. OG
* [Official course information](https://inside.i-med.ac.at/online/wbLv.wbShowLVDetail?pStpSpNr=872000)

## Lecturer
[Stefan Coassin](stefan.coassin@i-med.ac.at), [Sebastian Schoenherr](sebastian.schoenherr@i-med.ac.at)  
Institute of Genetic Epidemiology  
Medical University of Innsbruck 

## Timetable

### Day 1
Stefan, Sebastian:
* General / Sequencing Introduction

Stefan:
* NGS Technology Basics; Illumina; Iontorrent 
* Library types Single, Paired, mate 

Sebastian:
* Bioinformatics Introduction

*Lunch 

Sebastian:
* Linux Introduction 
* Install Tools and Trim
* [Exercises Day 1](scripts/linux-basics.md)

### Day 2 
Stefan:
* Data formats: FASTQ 
* Look at datasets & navigate through folders 
* Data QC: FASTQC, multiQC 
* Data QC: FASTQC, multiQC Exercises 

*Lunch 

Sebastian:
* Alignment
* [Alignment Exercises](https://github.com/seppinho/ngs-class/blob/master/scripts/mapping.md)
* Tablet Viewer, Variant Calling + Simple Exercise 
* [Exercises](https://github.com/seppinho/ngs-class/blob/master/scripts/variant-calling.md) 


### Day 3
Sebastian:
* Bash Scripting
* Introduction to Nextflow
* [Exercises Day 2](scripts/tool-installation.md) (skipped)


*Lunch

Stefan:
* Annotation, BCFTools and VCFtools

### Day 4
Stefan, Sebastian:
* Genepi Hackathon

## Credits
https://github.com/quinlan-lab/applied-computational-genomics
